Port Vala for Java Programmers Developer Guide

halfmexican opened the pull request:
Just drafting this PR. This is what I managed to get done today.

halfmexican committed:
add developer-guides dir

halfmexican committed:
Add first few pages

halfmexican committed:
Merge remote-tracking branch 'upstream/main'

halfmexican committed:
fix merge conflict

halfmexican committed:
add :numbered:

colinkiama commented:
So far so good!!

Before you continue though, since there's going to be a flat structure for this guide, I suggest renaming the files so they have this format `xx-filename.rst` starting from `01-source-file-and-compilation`.

halfmexican committed:
Merge branch 'main' into java-dev

halfmexican committed:
Port more pages

halfmexican committed:
Merge remote-tracking branch 'origin/java-dev'

halfmexican committed:
Fix index and edit pages

colinkiama committed:
Merge remote-tracking branch 'origin/main' into java-dev

colinkiama committed:
Restructure Java Developers guide to match the file structure of the other developer guides

halfmexican committed:
Merge branch 'vala-lang:main' into java-dev

colinkiama committed:
Merge branch 'java-dev' of github.com:halfmexican/vala-docs; branch 'main' of github.com:vala-lang/vala-docs into java-dev

colinkiama committed:
Convert existsing vala-for-java-devs pages to markdown

colinkiama committed:
Add Vala for Java Programmers guide to side bar and move valaodoc guide to top of documentation section

colinkiama committed:
Fix table in Java for devs strings table

halfmexican commented:
i'm alive and in the post-grad struggle. I wanna help the vala project. Gonna take a look back at this

colinkiama committed:
Merge branch 'main' into java-dev

Resolve conflicts in docs/developer-guides/index.md

colinkiama committed:
Port all Java for Vala Programmers documentation

- Update Java for Vala Programmers docs directory structure
- Organise each of the doc pages into chapters
- Update links to the new documentation structure

Fixes #148
